src/components/Dashboard/ThisWeek/NewsletterMenu/MenuItems/ExportPhysical/ExportPhysical.test.tsx

```diff
@@ -41,6 +52,121 @@ describe('ExportPhysical', () => {
         />
       </MockedProvider>,
     );
-    expect(queryByText('Export Contacts')).toBeInTheDocument();
+    expect(getByText('Export Contacts')).toBeInTheDocument();
+  });
+
+  describe('Exporting Contacts', () => {
+    const { location } = window;
+
+    beforeAll(() => {
+      // Have to ignore TS complaining about deleting window.location because location is not optional
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      delete window.location;
+      window.location = { ...location, replace: jest.fn() };
+    });
+
+    afterAll(() => {
+      window.location = location;
+    });
```

Comment:
I looked everywhere to find a way to mock ```window.location.replace``` and this was the only one that worked. However, a change in TypeScript 4 causes a TS issue with the ```delete window.location```. Let me know how you feel about this, or if you have any suggestions.

Reply:
https://stackoverflow.com/a/56999581/665224 is a possibility.

I also saw mention of using `global` instead of `window`. Not quite sure what there is to be gained from that.

With this solution, how come `delete window.location` is required? Does overwriting it throw errors?

Reply:
I think I tried that solution with no success, but I'll give it another shot. Overwriting doesn't throw errors, but it doesn't work. It appears you have to delete the existing one set by jsdom and then set it again but with your mock.